### ISSUE TITLE
Plugin Respond Issue

### ISSUE THREAD

**Issue by Jalaj1087** [asmpro7/ElementFlow#1]
[2024-07-17.txt](https://github.com/user-attachments/files/16260498/2024-07-17.txt)

Flow Launcher version: 1.18.0
OS Version: 19045.4651
IntPtr Length: 8
x64: True

Python Path: C:\Users\admin\AppData\Roaming\FlowLauncher\Environments\Python\PythonEmbeddable-v3.11.4\pythonw.exe
Node Path: 
Date: 07/17/2024 10:39:03
Exception:
ElementFlow Exception: 
Websites: https://github.com/asmpro7/ElementFlow
Author: Ahmed ElSaeed
Version: 1.0.4
Flow.Launcher.Core.ExternalPlugins.FlowPluginException: Traceback (most recent call last):
  File "C:\Users\admin\AppData\Roaming\FlowLauncher\Plugins\ElementFlow-1.0.4\lib\numpy\core\__init__.py", line 23, in <module>
    from . import multiarray
  File "C:\Users\admin\AppData\Roaming\FlowLauncher\Plugins\ElementFlow-1.0.4\lib\numpy\core\multiarray.py", line 10, in <module>
    from . import overrides
  File "C:\Users\admin\AppData\Roaming\FlowLauncher\Plugins\ElementFlow-1.0.4\lib\numpy\core\overrides.py", line 6, in <module>
    from numpy.core._multiarray_umath import (
ModuleNotFoundError: No module named 'numpy.core._multiarray_umath'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\admin\AppData\Roaming\FlowLauncher\Plugins\ElementFlow-1.0.4\main.py", line 13, in <module>
    import numpy as np
  File "C:\Users\admin\AppData\Roaming\FlowLauncher\Plugins\ElementFlow-1.0.4\lib\numpy\__init__.py", line 141, in <module>
    from . import core
  File "C:\Users\admin\AppData\Roaming\FlowLauncher\Plugins\ElementFlow-1.0.4\lib\numpy\core\__init__.py", line 49, in <module>
    raise ImportError(msg)
ImportError: 

IMPORTANT: PLEASE READ THIS FOR ADVICE ON HOW TO SOLVE THIS ISSUE!

Importing the numpy C-extensions failed. This error can happen for
many reasons, often due to issues with your setup or how NumPy was
installed.

We have compiled some common reasons and troubleshooting tips at:

    https://numpy.org/devdocs/user/troubleshooting-importerror.html

Please note and check the following:

  * The Python version is: Python3.11 from "C:\Users\admin\AppData\Roaming\FlowLauncher\Environments\Python\PythonEmbeddable-v3.11.4\pythonw.exe"
  * The NumPy version is: "1.24.4"

and make sure that they are the versions you expect.
Please carefully study the documentation linked above for further help.

Original error was: No module named 'numpy.core._multiarray_umath'

 ---> System.IO.InvalidDataException: Traceback (most recent call last):
  File "C:\Users\admin\AppData\Roaming\FlowLauncher\Plugins\ElementFlow-1.0.4\lib\numpy\core\__init__.py", line 23, in <module>
    from . import multiarray
  File "C:\Users\admin\AppData\Roaming\FlowLauncher\Plugins\ElementFlow-1.0.4\lib\numpy\core\multiarray.py", line 10, in <module>
    from . import overrides
  File "C:\Users\admin\AppData\Roaming\FlowLauncher\Plugins\ElementFlow-1.0.4\lib\numpy\core\overrides.py", line 6, in <module>
    from numpy.core._multiarray_umath import (
ModuleNotFoundError: No module named 'numpy.core._multiarray_umath'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\Users\admin\AppData\Roaming\FlowLauncher\Plugins\ElementFlow-1.0.4\main.py", line 13, in <module>
    import numpy as np
  File "C:\Users\admin\AppData\Roaming\FlowLauncher\Plugins\ElementFlow-1.0.4\lib\numpy\__init__.py", line 141, in <module>
    from . import core
  File "C:\Users\admin\AppData\Roaming\FlowLauncher\Plugins\ElementFlow-1.0.4\lib\numpy\core\__init__.py", line 49, in <module>
    raise ImportError(msg)
ImportError: 

IMPORTANT: PLEASE READ THIS FOR ADVICE ON HOW TO SOLVE THIS ISSUE!

Importing the numpy C-extensions failed. This error can happen for
many reasons, often due to issues with your setup or how NumPy was
installed.

We have compiled some common reasons and troubleshooting tips at:

    https://numpy.org/devdocs/user/troubleshooting-importerror.html

Please note and check the following:

  * The Python version is: Python3.11 from "C:\Users\admin\AppData\Roaming\FlowLauncher\Environments\Python\PythonEmbeddable-v3.11.4\pythonw.exe"
  * The NumPy version is: "1.24.4"

and make sure that they are the versions you expect.
Please carefully study the documentation linked above for further help.

Original error was: No module named 'numpy.core._multiarray_umath'

   at Flow.Launcher.Core.Plugin.JsonRPCPlugin.ExecuteAsync(ProcessStartInfo startInfo, CancellationToken token) in C:\projects\flow-launcher\Flow.Launcher.Core\Plugin\JsonRPCPlugin.cs:line 243
   at Flow.Launcher.Core.Plugin.JsonRPCPlugin.ExecuteAsync(ProcessStartInfo startInfo, CancellationToken token) in C:\projects\flow-launcher\Flow.Launcher.Core\Plugin\JsonRPCPlugin.cs:line 248
   at Flow.Launcher.Core.Plugin.JsonRPCPlugin.QueryAsync(Query query, CancellationToken token) in C:\projects\flow-launcher\Flow.Launcher.Core\Plugin\JsonRPCPlugin.cs:line 261
   at Flow.Launcher.Core.Plugin.PluginManager.<>c__DisplayClass23_0.<<QueryForPluginAsync>b__0>d.MoveNext() in C:\projects\flow-launcher\Flow.Launcher.Core\Plugin\PluginManager.cs:line 196
--- End of stack trace from previous location ---
   at Flow.Launcher.Infrastructure.Stopwatch.DebugAsync(String message, Func`1 action) in C:\projects\flow-launcher\Flow.Launcher.Infrastructure\Stopwatch.cs:line 34
   at Flow.Launcher.Core.Plugin.PluginManager.QueryForPluginAsync(PluginPair pair, Query query, CancellationToken token) in C:\projects\flow-launcher\Flow.Launcher.Core\Plugin\PluginManager.cs:line 195
   --- End of inner exception stack trace ---
   at Flow.Launcher.Core.Plugin.PluginManager.<>c__DisplayClass23_1.<QueryForPluginAsync>b__1(ActionContext _) in C:\projects\flow-launcher\Flow.Launcher.Core\Plugin\PluginManager.cs:line 224
   at Flow.Launcher.Plugin.Result.ExecuteAsync(ActionContext context) in /_/Flow.Launcher.Plugin/Result.cs:line 246
   at Flow.Launcher.ViewModel.MainViewModel.OpenResultAsync(String index) in C:\projects\flow-launcher\Flow.Launcher\ViewModel\MainViewModel.cs:line 307
   at CommunityToolkit.Mvvm.Input.AsyncRelayCommand.AwaitAndThrowIfFailed(Task executionTask)
   at System.Threading.Tasks.Task.<>c.<ThrowAsync>b__128_0(Object state)
   at System.Windows.Threading.ExceptionWrapper.InternalRealCall(Delegate callback, Object args, Int32 numArgs)
   at System.Windows.Threading.ExceptionWrapper.TryCatchWhen(Object source, Delegate callback, Object args, Int32 numArgs, Delegate catchHandler)

**Comment by lazycodeman** [comment]
Your error is stated here:
ModuleNotFoundError: No module named 'numpy.core._multiarray_umath'

In the folder python 3.11, a "numpy library" needs to be installed.

I suggest you install a Python version on your own (not auto installation by Flow Launch), then point the Python exe in the General tab setting in Flow Launcher.

Hope it help.

**Comment by Lucas256-cmd** [comment]
Hello I also made a plugin for looking up elements in the periodic table. It shows way less info but it also uses the same .json file or at least a file from the same repository. It worked pretty well but @Jalaj1087 requested to implement search by atomic number. I don't know why but I looked at his profile (normally I don't do this because it feels like I'm stalking someone but for some reason I checked it out) and found this issue, I was like huh another periodic table plugin, but I searched for this and never found one. It's because I was searching for periodic.... but never chemical....

So I found this plugin and tested it, I already knew it wouldn't work but tried anyway. The problem was numpy, an explanation is found in my pull request #2. After a search in the code I found numpy was only used twice and both were to create the image of the element color. When removing all code for the images, I found this plugin was way better than mine so I wanted to try and make it work with the build in python of flowlauncher.

Then I tried searching for solutions for the numpy problem. And guess what, the solution was completely no numpy, it is a library purely written in python: https://gitlab.com/drj11/pypng/-/tree/main?ref_type=heads. So I implemented a modified version of the image function and now it should work, at least according to my testing.

TLDR: This problem has a fix. And in my opinion a better one than installing python yourself. 

Edit: It's not the same json file :)
